### PR TITLE
stm32u5 supports the BackuUp sram

### LINKED ISSUE #54934: stm32u5, samples\boards\stm32\backup_sram not supported
**Describe the bug**
It seems that support for U5 series BKPSRAM is missing.

I'am using b_u585i_iot02a board. My goal is to enable BKPSRAM using sample backup_sram. U5 is missing from
kconfig file so I added
|| SOC_SERIES_STM32U5X.

"
warning: STM32_BACKUP_SRAM (defined at soc/arm/st_stm32\common\Kconfig.soc:12) was assigned the
value 'y' but got the value 'n'. Check these unsatisfied dependencies: (SOC_SERIES_STM32F2X ||
SOC_SERIES_STM32F4X || SOC_SERIES_STM32F7X || SOC_SERIES_STM32H7X)
"

I also added to app.overlay:

	soc {
			backup_sram: memory@40036400 {
			compatible = "zephyr,memory-region", "st,stm32-backup-sram";
			reg = <0x40036400 DT_SIZE_K(2)>;
			clocks = <&rcc STM32_CLOCK_BUS_AHB1 0x10020000>;  /* BKPSRAMEN and RAMCFGEN */
			zephyr,memory-region = "BACKUP_SRAM";
			status = "disabled";
		};
	};
	
...

&backup_sram{
	status = "okay";
};

And now sample works. 

Also it would be nice to have driver support for BKPSRAM erase,
currently I'am using direct register writes to do it.	

**To Reproduce**
1. west build -p auto -b b_u585i_iot02a_ns samples\boards\stm32\backup_sram

**Expected behavior**
build should be ok

**Impact**
medium

**Environment (please complete the following information):**
 - OS: Windows

### PR DESCRIPTION
**Pull request by FRASTM**
Add the support of the BackUp SRAM  to the stm32u5.
Make the  samples/boards/stm32/backup_sram  run on stm32u585 disco or stm32u575 nucleo boards.

Note that there is no reset register from the RCC to the BackUp Ram

FIxes https://github.com/zephyrproject-rtos/zephyr/issues/54934

Signed-off-by: Francois Ramu <francois.ramu@st.com>

### CODE REVIEW

**Comment by FRASTM** [comment]
When the flag is set : 
```
*** Booting Zephyr OS build v3.3.0-rc3-7-g60f9ccaa781a ***
Current value in backup SRAM (0x40036400): 0
Next reported value should be: 1
Keep VBAT power source and reset the board now!
*** Booting Zephyr OS build v3.3.0-rc3-7-g60f9ccaa781a ***
Current value in backup SRAM (0x40036400): 0
Next reported value should be: 1
Keep VBAT power source and reset the board now!
```
When the flag is reset : 
```
*** Booting Zephyr OS build v3.3.0-rc3-7-g60f9ccaa781a ***
Current value in backup SRAM (0x40036400): 0
Next reported value should be: 1
Keep VBAT power source and reset the board now!
*** Booting Zephyr OS build v3.3.0-rc3-7-g60f9ccaa781a ***
Current value in backup SRAM (0x40036400): 1
Next reported value should be: 2
Keep VBAT power source and reset the board now!
*** Booting Zephyr OS build v3.3.0-rc3-7-g60f9ccaa781a ***
Current value in backup SRAM (0x40036400): 2
Next reported value should be: 3
Keep VBAT power source and reset the board now!
```

**Review by erwango — CHANGES_REQUESTED** [review]
I'm globally fine with the change, but I would remove `CONFIG_ERASE_BKP_SRAM_AT_BOOT` support. This option would anyway disappear if driver is moved to bbram/nvm driver

File: soc/arm/st_stm32/common/Kconfig.soc [hunk 1]
```diff
@@ -12,10 +12,17 @@ config STM32_CCM
 config STM32_BACKUP_SRAM
 	bool "STM32 Backup SRAM"
 	depends on SOC_SERIES_STM32F2X || SOC_SERIES_STM32F4X || \
-		   SOC_SERIES_STM32F7X || SOC_SERIES_STM32H7X
+		   SOC_SERIES_STM32F7X || SOC_SERIES_STM32H7X || \
+		   SOC_SERIES_STM32U5X
```
[the remark this answers is not in the document]
Comment (FRASTM) [reply soc/arm/st_stm32/common/Kconfig.soc]: list of soc series replaced by DT_HAS_ST_STM32_BACKUP_SRAM_ENABLED

**Comment by FRASTM** [comment]
rebase
remove the erase part

**Review by erwango — COMMENTED** [review]
Please update PR description (clean the erase option part)
Otherwise LGTM

**Review by erwango — APPROVED** [review]

**Review by gautierg-st — APPROVED** [review]

**Pull request merged**